### ISSUE TITLE
Not really a bug; but gameplay issue.

### ISSUE THREAD

**Issue by Ziabatsu** [HelloFangaming/HelloMarioEngine#11]
There are 3 bug/gameplay inconsistencies I've noticed that are more blatant than not.

When I try to jump on a turtle, the hit box on the turtle usually kills me instead of killing it (which I've never had a problem doing in Mario before).

When I beat a level, it shows a red "m" for mario showing I beat the level, then the marker is immediately removed.

When I run, the deceleration process doesn't happen quickly enough. I can get a running start and just hit the jump button a bunch of times (not even holding forward once I sprint and do my first jump) to jump across a large portion of the level because my momentum doesn't really stop quickly at all.

These are my initial findings. I'll report more if I find anything :)

**Comment by HelloFangaming** [comment]
1) By turtle, are you talking about regular koopas (obj_koopa), because those seem to work fine for me. Their hit box is the same size as other small enemies like goombas, for example.

2) Are there any things I have to do before beating the level to make this bug happen, because this isn't happening to me...

3) Actually, the official Mario games work this way too.

**Comment by Ziabatsu** [comment]
1). The Green Coopa Turtle. When I try to jump on him, 80% of the time, I
manage to get killed instead of pop him out of his shell; where in the SNES
version he would have popped out of the shell. It's the very first thing I
noticed while actually playing. I think I died like 10 times trying to jump
him out of the shell? I kept trying it at an angle, but it was still right
in the area that should have killed it.

2.) I watched a mario3 speedrun because I started to doubt my own
complaint; nothing wrong here after all. It was when you killed the hammer
bro's; a "m" marker appears then goes away, but it's supposed to just like
in SMB3. Sorry about that.

3.) Fair enough. For w/e reason I felt it was a little floaty/speedy, but
it might just be my imagination.
